File: DotNet Project V2.0/backend-dotnet/src/Emart.Domain/Entities/SubCategory.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Emart.Domain.Entities;

[Table("subcategory_master")]
public class SubCategory
{
    [Key]
    [Column("subcat_master_id")]
    public int SubcatMasterId { get; set; }

    [Column("subcat_id")]
    public string? SubcatId { get; set; }

    [Column("subcat_name")]
    public string? SubcatName { get; set; }

    [Column("subcat_image_path")]
    public string? SubcatImagePath { get; set; }

    [Column("flag")]
    public bool? Flag { get; set; }

    [Column("catmaster_id")]
    public int? CatmasterId { get; set; }

    public Category? Category { get; set; }
}

File: Java Project  V1.7/backend-spring/src/main/java/com/emart/entity/SubCategory.java
package com.emart.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.*;

@Entity
@Table(name = "subcategory_master")
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class SubCategory 
{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer subcatMasterId;

    private String subcatId;

    private String subcatName;

    private String subcatImagePath;

    private Boolean flag;


    @ManyToOne
    @JoinColumn(name = "catmaster_id")
    private Category category;


    public Integer getSubcatMasterId() 
    {
        return subcatMasterId;
    }

    public void setSubcatMasterId(Integer subcatMasterId) 
    {
        this.subcatMasterId = subcatMasterId;
    }


    public String getSubcatId() 
    {
        return subcatId;
    }

    public void setSubcatId(String subcatId) 
    {
        this.subcatId = subcatId;
    }


    public String getSubcatName() 
    {
        return subcatName;
    }

    public void setSubcatName(String subcatName) 
    {
        this.subcatName = subcatName;
    }


    public String getSubcatImagePath() 
    {
        return subcatImagePath;
    }

    public void setSubcatImagePath(String subcatImagePath)
    {
        this.subcatImagePath = subcatImagePath;
    }


    public Boolean getFlag()
    {
        return flag;
    }

    public void setFlag(Boolean flag) 
    {
        this.flag = flag;
    }


    public Category getCategory() 
    {
        return category;
    }

    public void setCategory(Category category) 
    {
        this.category = category;
    }
}